pkg/api/unversioned.go

```diff
@@ -24,14 +24,33 @@ import (
 
 // APIVersions lists the api versions that are available, to allow
 // version negotiation. APIVersions isn't just an unnamed array of
-// strings in order to allow for future evolution, though unversioned
+// strings in order to allow for future evolution, though unversioned.
 type APIVersions struct {
+	// Versions are the api versions that are available.
 	Versions []string `json:"versions"`
 }
 
+// APIResource stores the name of a resource and whether it is namespaced.
+type APIResource struct {
+	// Resource is the name of the resource.
+	Resource string `json:"resource"`
+	// Namespaced indicates if a resource is namespaced or not.
+	Namespaced bool `json:"namespaced"`
+}
+
+// APIResourceList is a list of APIResource, it is used to store the name of the
+// resources supported in a specific group and version, and whether the resource
+// is namespaced.
+type APIResourceList struct {
+	// APIResources store the name of the resources and whether they are
+	// namespaced.
+	APIResources []APIResource `json:"resources"`
```

Comment:
@bgrant0607 Would it be useful to add `Version string` as a field here? I know the client should already know that since it's in the URL path, but somehow it seems like it'd be nice to repeat it for them here.

Alternatively, it could go in the APIResource struct but that'd be very repetitive.